publication-commons/src/main/java/no/unit/nva/publication/model/PublicationSummary.java

```diff
@@ -34,21 +41,39 @@ public class PublicationSummary {
     private Instant modifiedDate;
     @JsonProperty
     private PublicationStatus status;
-    
+    @JsonProperty
+    private PublicationInstance<? extends Pages> publicationInstance;
+    @JsonProperty
+    private Instant publishedDate;
+    @JsonProperty
+    private List<Contributor> contributors;
+
     public static PublicationSummary create(Publication publication) {
         var publicationSummary = new PublicationSummary();
         publicationSummary.setIdentifier(publication.getIdentifier());
         publicationSummary.setPublicationId(toPublicationId(publication.getIdentifier()));
         publicationSummary.setCreatedDate(publication.getCreatedDate());
         publicationSummary.setModifiedDate(publication.getModifiedDate());
+        publicationSummary.setPublishedDate(publication.getPublishedDate());
         publicationSummary.setOwner(new User(publication.getResourceOwner().getOwner().getValue()));
         publicationSummary.setStatus(publication.getStatus());
-        if (nonNull(publication.getEntityDescription())) {
-            publicationSummary.setTitle(publication.getEntityDescription().getMainTitle());
+        var entityDescription = publication.getEntityDescription();
+        if (nonNull(entityDescription)) {
+            publicationSummary.setTitle(entityDescription.getMainTitle());
+            if (nonNull(entityDescription.getReference())) {
+                publicationSummary.setPublicationInstance(entityDescription.getReference().getPublicationInstance());
+            }
+            if (nonNull(entityDescription.getContributors())) {
+                publicationSummary.setContributors(entityDescription.getContributors()
+                                                       .stream()
+                                                       .sorted(Comparator.comparing(Contributor::getSequence))
+                                                       .limit(MAX_SIZE_CONTRIBUTOR_LIST)
+                                                       .collect(Collectors.toList()));
+            }
```

Comment:
Instead of having multiple null checks in create() method you can extract all that logic, as you are doing with identifier extractPublicationIdentifier(publicationId).

public static PublicationSummary create(Publication publication) {
        var publicationSummary = new PublicationSummary();
        publicationSummary.setIdentifier(publication.getIdentifier());
        publicationSummary.setPublicationId(toPublicationId(publication.getIdentifier()));
        publicationSummary.setCreatedDate(publication.getCreatedDate());
        publicationSummary.setModifiedDate(publication.getModifiedDate());
        publicationSummary.setPublishedDate(publication.getPublishedDate());
        publicationSummary.setOwner(new User(publication.getResourceOwner().getOwner().getValue()));
        publicationSummary.setStatus(publication.getStatus());
        publicationSummary.setTitle(extractTitle(publication));
        publicationSummary.setPublicationInstance(extractPublicationInstance(publication));
        publicationSummary.setContributors(extractContributors(publication));
        return publicationSummary;
    }

    private static List<Contributor> extractContributors(Publication publication) {
        return Optional.of(publication.getEntityDescription())
                .map(EntityDescription::getContributors)
                .stream()
                .flatMap(Collection::stream)
                .sorted(Comparator.comparing(Contributor::getSequence))
                .limit(MAX_SIZE_CONTRIBUTOR_LIST)
                .collect(Collectors.toList());
    }

    private static PublicationInstance<? extends Pages> extractPublicationInstance(Publication publication) {
        return Optional.ofNullable(publication.getEntityDescription())
                .map(EntityDescription::getReference)
                .map(Reference::getPublicationInstance)
                .orElse(null);
    }

    public static String extractTitle(Publication publication) {
        return Optional.ofNullable(publication.getEntityDescription())
                .map(EntityDescription::getMainTitle)
                .orElse(null);
    }

Reply:
Thanks, will do!